### PR TITLE
fix for case 175213

### PR DESCRIPTION
**Pull request by lwyszomi**
@sravfeyn @esoergel

### CODE REVIEW

**Comment by czue** [comment]
build fail looks real:
```
======================================================================
FAIL: test_vhnd_before_checkpoint_pregnancy (custom.opm.tests.test_windows_and_months.TestChildFirstMonthWindow)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/custom/opm/tests/test_windows_and_months.py", line 167, in test_vhnd_before_checkpoint_pregnancy
    self.assertRaises(InvalidRow, MockCaseRow, case, self.report, data_provider)
AssertionError: InvalidRow not raised
----------------------------------------------------------------------
Ran 611 tests in 57.999s
```

File: custom/opm/tests/test_status.py [hunk 1]
```diff
@@ -57,4 +59,6 @@ def test_due_after_period_not_delivered(self):
             forms=[],
             edd=date(2014, 5, 10),
         )
-        self.assertRaises(InvalidRow, MockCaseRow, case, self.report)
+        mock_case = MockCaseRow(case, self.report)
+        self.assertTrue(mock_case.case_is_out_of_range)
+
```
Comment (dimagimon) [inline custom/opm/tests/test_status.py]: W391 blank line at end of file

**Comment by lwyszomi** [comment]
@czue @esoergel @sravfeyn Updated

**Comment by lwyszomi** [comment]
changes for cases:
http://manage.dimagi.com/default.asp?175213#970232
http://manage.dimagi.com/default.asp?175312#970229

File: custom/opm/reports.py [hunk 2]
```diff
@@ -892,14 +891,19 @@ def row_objects(self):
...
                 case_row = self.get_row_data(row, index=1, awc_codes=awc_codes)
-                total_payment += case_row.cash_amt
-                rows.append(case_row)
-            except CaseOutOfRange:
-                try:
-                    rows.append(FakeConditionsMet(row, self, child_index=1, awc_codes=awc_codes))
-                except InvalidRow as e:
-                    if self.debug:
-                        self.add_debug_data(row._id, e)
+                if not case_row.case_is_out_of_range:
```
Comment (esoergel) [inline custom/opm/reports.py]: Have you verified that this doesn't change the BPR or CMR?  Not everything is called on `__init__`, so if a property is accessed later which sets the flag to `True`, this check won't catch it.  Is it possible to move this until after all the calculations have happened?

Comment (lwyszomi) [inline custom/opm/reports.py]: @esoergel I exclude invalid rows in other reports.

I checked all code for reports and method ```get_row_data``` returns us the whole object and all properties calculated. If not we will get a traceback in method ```rows``` when we use specific property to get data. But now we only catch the InvalidRow  in ```row_objects```.

**Comment by sravfeyn** [comment]
hey @esoergel can you merge this once you have reviewed today (anushree is trying to get this to OPM)

**Comment by czue** [comment]
whoever merges this - make sure you revert https://github.com/dimagi/commcare-hq/pull/7348

**Comment by esoergel** [comment]
Merged and reverted https://github.com/dimagi/commcare-hq/pull/7348 (https://github.com/dimagi/commcare-hq/pull/7356)

**Pull request merged**